compute/cker/src/train/BinaryArithmetic.test.cc

```diff
@@ -217,8 +217,78 @@ TEST(CKer_Operation, MulGrad)
   }
 }
 
-TEST(CKer_Operation, DistinctShape)
+TEST(CKer_Operation, neg_DistinctShape)
 {
+  {
+    // all but lhs have the same shape
+    std::vector<float> lhs(8); // Does not matter
+    std::vector<float> rhs(6); // Does not matter
+    std::vector<float> incoming_backward = {-2, 3, -4, 5, -6, 7};
+    std::vector<float> lhs_backward(6);
+    std::vector<float> rhs_backward(6);
+
+    EXPECT_ANY_THROW(nnfw::cker::train::BinaryArithmeticGrad(
+      nnfw::cker::Shape{2, 2, 2}, lhs.data(), nnfw::cker::Shape{2, 3}, rhs.data(),
+      nnfw::cker::Shape{2, 3}, incoming_backward.data(), nnfw::cker::Shape{2, 3},
+      lhs_backward.data(), nnfw::cker::Shape{2, 3}, rhs_backward.data(),
+      nnfw::cker::train::ArithmeticType::kAdd));
```

Comment:
Note that `ArithmeticType` does not affect the shape check.

https://github.com/Samsung/ONE/blob/e516966204432357fb10547c0bf24fc238b01575/compute/cker/include/cker/train/operation/BinaryArithmetic.h#L39-L47